File: solutions/_24_swap_nodes_in_pairs/solution.py
from typing import Optional


# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head or not head.next:
            return head
        a = head
        b = head.next
        a.next = self.swapPairs(b.next)
        b.next = a
        return b


# https://leetcode.com/problems/swap-nodes-in-pairs/

File: solutions/_24_swap_nodes_in_pairs/Solution.java

package _24_swap_nodes_in_pairs;

class Solution {
    public ListNode swapPairs(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        ListNode a = head;
        ListNode b = head.next;
        a.next = swapPairs(b.next);
        b.next = a;
        return b;
    }
}

/*
 * https://leetcode.com/problems/swap-nodes-in-pairs/
 */